gamemodes/terrortown/gamemode/shared/sh_player_ext.lua

```diff
@@ -1261,28 +1261,80 @@ end
 -- @param number fov The angle of perception (FOV); set to 0 to return to default user FOV
 -- @param[default=0] number time The time it takes to transition to the FOV expressed in a floating point
 -- @param[default=self] Entity requester The requester or "owner" of the zoom event; only this entity will be able to change the player's FOV until it is set back to 0
--- @param[default=false] boolean isSprinting Whether this is called to set sprinting FOV
 -- @realm shared
-function plymeta:SetFOV(fov, time, requester, isSprinting)
-    if isSprinting then
-        self.sprintingFOV = fov
+function plymeta:SetFOV(fov, time, requester)
+    -- if dynamic FOV is disabled, the default function should be used
+    if not self:GetPlayerSetting("enable_dynamic_fov") then
+        self:SetOldFOV(fov, time, requester)
 
-        if self.externalFOV then
-            return
-        end
+        return
     end
 
-    if not isSprinting then
-        self.externalFOV = fov
+    -- normal case, SetFOV is used to set a fixed FOV value, we should track
+    -- this so we know that we should no apply custom FOV in GM:CalcView
+    if fov and fov ~= 0 then
+        self.fixedFOV = true
+        self.triggeredFOV = false
+        self.timeFOV = time
 
-        if fov == 0 then
-            self.externalFOV = nil
+        if SERVER then
+            net.Start("TTT2UpdateFOV")
+            net.WriteEntity(self)
+            net.WriteBool(true) -- FOV is fixed now
+            net.WriteFloat(time)
+            net.WriteFloat(self:GetFOV())
+            net.Broadcast()
+        end
 
-            fov = self.sprintingFOV or 0
+    -- when we zoom out, this has to trigger our custom zoom out that handles different
+    -- desired FOV values instead of 0, we therefore set it without transition to handle
+    -- the transistion ourself
+    else
+        -- the SetFOV function is called multiple times, this check makes sure
+        -- we only trigger on the first call after a zoom out
+        if self.fixedFOV then
+            self.fixedFOV = false
+            self.triggeredFOV = true
+            self.timeFOV = time
+
+            if SERVER then
+                net.Start("TTT2UpdateFOV")
+                net.WriteEntity(self)
+                net.WriteBool(false) -- FOV isn't fixed anymore
+                net.WriteFloat(time)
+                net.WriteFloat(self:GetFOV())
+                net.Broadcast()
+            end
         end
+
+        -- set time to 0 so our custom FOV code can handle the zoom out
+        time = 0
     end
 
-    self:SetOldFOV(fov, time, requester, isSprinting)
+    self:SetOldFOV(fov, time, requester)
+end
+
+if CLIENT then
+    -- we have to update this via netmessage as well, because some addons only call
+    -- the SetFOV on the server, so we have to manually sync it
+    net.Receive("TTT2UpdateFOV", function()
+        local ply = net.ReadEntity()
```

Comment:
I think we can remove this, as each client only needs to set this for themselves, right?

Reply:
I intended to support it as a spec as well, but I had no means to test it and it appears to be doing nothing right now. So I can remove it for now and readd it in a PR when I properly test it for specs

Reply:
We have to test it today in our group session, but removing this might cause issues when spectating. So I'd like to keep it as a broadcast